lib/bunq/signature.rb

```diff
@@ -30,6 +30,10 @@ def verify!(response)
       sorted_bunq_headers = response.raw_headers.select(&method(:verifiable_header?)).sort.to_h.map { |k, v| "#{k.to_s.split('-').map(&:capitalize).join('-')}: #{v.first}" }
       data = %Q{#{response.code}\n#{sorted_bunq_headers.join("\n")}\n\n#{response.body}}
 
+      if response.code == 409 || response.code == 429
```

Comment:
I think a better place to implement this is the `Bunq::Resource.handle_response`. The reason is the signature verification can be disabled (it is optional) and then this error is never raised. 
Also the `post` of Installations must skips the verify since there is no certificate installed yet.

But it in case it is enabled we want to verify before handling the response. So I think this method should early exit when the response code is `409` or `429`. The handle response can then fail with the correct errors, and maybe other errors that are raised for which the response can be verified. This way error handling is (sort of) in a single location.

Last remark: since `409` is not documented in https://doc.bunq.com/api/1/page/errors I would rather make the check more specific in: `Bunq::Client.configuration.sandbox && code == 409`

Reply:
✅  done :)